x-pack/plugins/actions/server/unsecured_actions_client/unsecured_actions_client.ts

```diff
@@ -54,25 +101,48 @@ export class UnsecuredActionsClient {
         `"${requesterId}" feature is not allow-listed for UnsecuredActionsClient access.`
       );
     }
+
     // Inject source based on requesterId
-    return this.executionEnqueuer(
-      this.internalSavedObjectsRepository,
-      this.injectSource(requesterId, actionsToExecute)
-    );
+    const actionsToEnqueue = actionsToExecute.map((action) => {
+      const source = this.getSourceFromRequester(
+        requesterId,
+        action.id,
+        action.relatedSavedObjects
+      );
+      return {
+        ...action,
+        ...source,
+      };
+    });
+    return this.executionEnqueuer(this.internalSavedObjectsRepository, actionsToEnqueue);
   }
 
-  private injectSource(requesterId: string, actionsToExecute: ExecuteOptions[]): ExecuteOptions[] {
+  private getSourceFromRequester(
+    requesterId: string,
+    actionId: string,
+    relatedSavedObjects?: RelatedSavedObjects
+  ) {
     switch (requesterId) {
       case NOTIFICATION_REQUESTER_ID:
-        return actionsToExecute.map((actionToExecute) => ({
-          ...actionToExecute,
+        return {
           source: asNotificationExecutionSource({
             requesterId,
-            connectorId: actionToExecute.id,
+            connectorId: actionId,
           }),
-        }));
+        };
+      case BACKGROUND_TASK_REQUESTER_ID:
+        const taskSO = (relatedSavedObjects ?? []).find((rso) => rso.type === 'task');
```

Comment:
Seems like we're dependent on the caller to add the task to the relatedSavedObjects, to get it in the source, but allow it to not be set as well.  Can we enforce this somehow?  Maybe even just logging a warning when it happens would be "good enough" ...

Reply:
Added a warning log in https://github.com/elastic/kibana/pull/178682/commits/c24ebb0c09a766b3f706135a525079ec4a46d0cc. I considered making it required but `relatedSavedObjects` is optional everywhere else so it didn't feel right to make it required for something that's used primarily for debugging/traceability.